agnpy/compton/synchrotron_self_compton.py

```diff
@@ -33,6 +33,40 @@ def __init__(self, blob, ssa=False, integrator=np.trapz):
         self.ssa = ssa
         self.integrator = integrator
 
+    def electron_energy_loss_rate_thomson(self, gamma):
+        fixed = ((4 / 3) * sigma_T * c * self.blob.u_ph_synch).to("erg/s")
+        return fixed * gamma ** 2
+
+    def electron_energy_loss_rate(self, gamma):
+        epsilon0 = nu_to_epsilon_prime(nu_to_integrate, self.blob.z, self.blob.delta_D)
+        sed_flux_synch = Synchrotron(self.blob).evaluate_sed_flux(
+            epsilon0,
+            self.blob.z,
+            self.blob.d_L,
+            self.blob.delta_D,
+            self.blob.B,
+            self.blob.R_b,
+            self.blob.n_e,
+            *self.blob.n_e.parameters,
+            gamma=gamma
+        )
+
+        # differential energy density of photons
+        u_epsilon0 = (3 * np.power(self.blob.d_L, 2) * sed_flux_synch) / (
+                c * np.power(self.blob.R_b, 2) * np.power(self.blob.delta_D, 4) * epsilon0
+        )
+        # factor 3 / 4 for averaging in a sphere - see comment inside the evaluate_sed_flux method for u_synch calculation
+        u_epsilon0 *= 3 / 4
+
+        # formulas from paragraph 2.1, Moderski2005
+        b_ = 4 * gamma.reshape(len(gamma),1) * epsilon0.value.reshape(1,len(epsilon0)) # matrix [gamma x epsilon]
+        b_flattened = b_.reshape(b_.shape[0] * b_.shape[1])
+        f_KN_flattened = np.array([1 / (1 + b) ** 1.5 if b < 1e4 else (9 / (2 * b**2)) * (log(b) - 11/6) for b in b_flattened])
+        f_KN = f_KN_flattened.reshape(len(gamma), len(epsilon0)) # back to [gamma x epsilon] matrix
```

Comment:
I think this double reshape and for loop also adds to the execution time of the code. You can do it simpler and probably faster using np.where and operating directly on the matrix